### PR TITLE
DOC: some 0.24.0 whatsnew clean-up

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Did some proofreading on the train

cc @TomAugspurger

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.rst [hunk 1]
```diff
@@ -142,14 +150,13 @@ See :ref:`Dtypes <basics.dtypes>` and :ref:`Attributes and Underlying Data <basi
...
-^^^^^
+``pandas.array``: a new top-level method for creating arrays
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A new top-level method :func:`array` has been added for creating 1-dimensional arrays (:issue:`22860`).
 This can be used to create any :ref:`extension array <extending.extension-types>`, including
-extension arrays registered by :ref:`3rd party libraries <ecosystem.extensions>`. See
-
-See :ref:`Dtypes <basics.dtypes>` for more on extension arrays.
+extension arrays registered by :ref:`3rd party libraries <ecosystem.extensions>`.
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v0.24.0.rst]: Trailing whitespace

**Review by TomAugspurger — APPROVED** [review]
Anything else you wanted for this @jorisvandenbossche?

File: doc/source/whatsnew/v0.24.0.rst [hunk 2]
```diff
@@ -69,7 +76,7 @@ These dtypes can operate as part of of ``DataFrame``.
    df.dtypes
 
 
-These dtypes can be merged & reshaped & casted.
+These dtypes can be merged, reshaped and casted.
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.24.0.rst]: I prefer the Oxford comma (instead of none).  Anyone else?

File: doc/source/whatsnew/v0.24.0.rst [hunk 3]
```diff
@@ -1177,7 +1168,7 @@ a default ``__eq__`` and ``__hash__``. If you have a parametrized dtype, you sho
 update the ``ExtensionDtype._metadata`` tuple to match the signature of your
 ``__init__`` method. See :class:`pandas.api.extensions.ExtensionDtype` for more (:issue:`22476`).
```
Comment (h-vetinari) [inline doc/source/whatsnew/v0.24.0.rst]: At the end of #22476, @TomAugspurger  and I were talking about an ambiguous sentence in the EA whatsnew: "Pandas now requires that extension dtypes be hashable."

How about: Pandas now requires that extension dtypes be hashable *(i.e. the respective `ExtensionDtype`-objects; hashability is not a requirement for the values of the corresponding `ExtensionArray`)*.

**Pull request merged**